File: Definitions/Game_Definitions/classes/HerculesProjectile.lua
---@meta
---@diagnostic disable

---@class HerculesProjectile : ExplodingBullet
HerculesProjectile = {}

---@return HerculesProjectile
function HerculesProjectile.new() return end

---@param props table
---@return HerculesProjectile
function HerculesProjectile.new(props) return end

---@param eventData gameprojectileShootEvent
---@return Bool
function HerculesProjectile:OnShoot(eventData) return end

---@param eventData gameprojectileShootTargetEvent
---@return Bool
function HerculesProjectile:OnShootTarget(eventData) return end

File: DecompiledGameScripts/cyberpunk/projectiles/herculesProjectile.swift

public class HerculesProjectile extends ExplodingBullet {

  protected cb func OnShoot(eventData: ref<gameprojectileShootEvent>) -> Bool {
    this.OnShootTarget(eventData as gameprojectileShootTargetEvent);
  }

  protected cb func OnShootTarget(eventData: ref<gameprojectileShootTargetEvent>) -> Bool {
    let energyLossFactorAfterCollision: Float;
    let gravitySimulation: Float;
    let startVelocity: Float;
    this.GeneralLaunchSetup(eventData);
    gravitySimulation = this.GetProjectileTweakDBFloatParameter("gravitySimulation");
    startVelocity = this.GetProjectileTweakDBFloatParameter("startVelocity");
    energyLossFactorAfterCollision = this.GetProjectileTweakDBFloatParameter("energyLossFactorAfterCollision");
    this.ParabolicLaunch(eventData, gravitySimulation, startVelocity, energyLossFactorAfterCollision);
  }
}
